src/options/options.js

```diff
@@ -112,6 +112,27 @@ async function setupListeners() {
       }
     }
   });
+
+  document.querySelectorAll(".menuItemAction").forEach(item => {
+    item.addEventListener("change", (event) => {
+      if (!checkMenuItemVisibility()) {
+        event.target.checked = true;
+      }
+    }
+    );
+  });
+}
```

Comment:
I think you could add this to the change options listener we already have and check the class name. Events will bubble up, so we don't need one listener for each element.

Reply:
You're right, it works.